FindYourMovie/Network/MovieManager.swift

```diff
@@ -0,0 +1,115 @@
+    ///
+    /// - Parameters:
+    ///     - query: Name of the movie to find.
+    ///     - page: String indicating which page number to fetch
+    ///     - completion: A Result object is return.
+    ///     -  result: On  success, a pair of movie results and page number is returned. On failure, a error enum is returned
+    func getMoviesNames(query : String, page : String, completion: @escaping(_ result: Result<MovieResult,MovieManagerError>) -> Void) {
+         
+        let getURL = "https://api.themoviedb.org/3/search/movie?api_key=\(MovieManager.API_KEY)&query=\(query)&page=\(page)"
+        
+        guard let url = URL(string: getURL) else {
+            DispatchQueue.main.async {
+                completion(.failure(.errorRequestingData(errorMessage: "Invalid URL")))
+            }
+            return
+        }
+        let dataTask = URLSession.shared.dataTask(with: url) {data, response, error in
+            
+            if let error = error {
+                DispatchQueue.main.async {
+                    completion(.failure(.errorRequestingData(errorMessage: error.localizedDescription)))
+                }
+            } else if
+                let jsonData = data,
+                let response = response as? HTTPURLResponse,
+                response.statusCode == 200 {
+                do {
+                    let decoder = JSONDecoder()
+                    let movieResponse = try decoder.decode(MovieResponse.self, from: jsonData)
+                    var movieResults = movieResponse.results
+                    if movieResponse.results.isEmpty {
+                        DispatchQueue.main.async {
+                            completion(.failure(.movieNotFound(errorMessage: "Search returned no results")))
+                        }
+                    } else {
+                        self.setMoviePosterPath(movies: &movieResults)
+                        DispatchQueue.main.async {
+                            completion(.success((movieResults, movieResponse.total_pages)))
+                        }
+                    }
+                    
+                } catch {
+                    DispatchQueue.main.async {
+                        completion(.failure(.errorProcessingData(errorMessage: error.localizedDescription)))
+                    }
+                    
+                }
+            }
+        }
+        dataTask.resume()
+    }
+    
+    /// Add base URL to movies by mutating the pposter field of a movie.
+    ///
+    /// - Parameters:
+    ///     - movies: List of fectched movies
+    ///
+    /// - Important: The poster field of the movie gets mutated
+    func setMoviePosterPath(movies :inout [Movie]) {
+        for (index, movie) in movies.enumerated() {
+            if movies[index].poster_path != nil {
+                movies[index].poster_path = "\(MovieManager.BASE_IMAGE_URL)\(movie.poster_path!)"
```

Comment:
The following is nicer because we remove a level of nesting by using a guard as well as avoiding any force unwrapping. In addition, the movie variable was never used so we do not need it. In general, avoid force unwrapping

```
for (index, _) in movies.enumerated() {
       guard let posterPath = movies[index].poster_path else { break }
       movies[index].poster_path = "\(MovieManager.BASE_IMAGE_URL)\(posterPath)"
}
```